exporter/elasticsearchexporter/mapping_hint.go

```diff
@@ -0,0 +1,31 @@
+package elasticsearchexporter
+
+import "go.opentelemetry.io/collector/pdata/pcommon"
+
+const (
+	mappingHintsAttrKey = "elasticsearch.mapping.hints"
+)
+
+type mappingHint string
+
+const (
+	hintAggregateMetricDouble mappingHint = "aggregate_metric_double"
+	hintDocCount              mappingHint = "_doc_count"
+)
+
+func hasHint(attrs pcommon.Map, hint mappingHint) (found bool) {
```

Comment:
This pattern is forcing us to iterate over the attribute every time we want to process a hint. I would prefer either processing all the hints together OR if that is not possible then extracting the hints in a separate struct and using it for checks whenever required.

Reply:
good point. I'll see how much extra effort is needed to preprocess to extract hints.

Reply:
done. I haven't gone so far to use a bitset, but i believe slices.Contains should be efficient enough.